### PR TITLE
Connected link

### PR DESCRIPTION
**Pull request by nathf**
Dispatch event on specific link clicks to focus heading on next screen, doesnt feel good.

### CODE REVIEW

**Comment by matt-sm** [comment]
You may need to show me why you need to do this

File: src/bundles/SellerRegistration/components/Finish.js [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React from 'react';
 
 const Finish = ({onClick}) => (
     <div>
-        <h1>That’s it for now!</h1>
+        <h1 tabIndex="-1">That’s it for now!</h1>
```
Comment (elisechant) [inline src/bundles/SellerRegistration/components/Finish.js]: i think you need to describe intent here. maybe the h1 should be wrapped in a hoc from the same file as connectedlink?

File: src/bundles/SellerRegistration/redux/helpers.js [hunk 2]
```diff
@@ -92,4 +92,14 @@ export const findDirtyForms = (state = {}) => {
       valid[key] = true;
       return valid;
     }, {});
+}
+
+// Not the greatest thing, but since we're not mutating anything
+// in the DOM and just managing focus, I think we're okay.
+// findDOMNode doesn't break the lines of encapsulation.
+export const focusHeading = () => {
```
Comment (elisechant) [inline src/bundles/SellerRegistration/redux/helpers.js]: yeah this should get wrapped up with your h1 component _ideally_

File: src/bundles/CaseStudy/components/DomainList/DomainList.js [hunk 3]
```diff
@@ -11,6 +11,9 @@ import ErrorBox       from '../../../../shared/form/ErrorBox';
 import StatefulError  from '../../../../shared/form/StatefulError';
 import formProps      from '../../../../shared/reduxModules/formPropsSelector';
 
+import ConnectedLink from '../../../SellerRegistration/components/ConnectedLink';
+import { linkClick } from '../../../SellerRegistration/redux/modules/application';
```
Comment (elisechant) [inline src/bundles/CaseStudy/components/DomainList/DomainList.js]: i think this view needs a container?

**Comment by elisechant** [comment]
@nathf maybe you could consider a factory pattern for Connected Links as props?

File: src/bundles/SellerRegistration/components/Finish.js (hunk 1 above)
Comment (nathf) [inline src/bundles/SellerRegistration/components/Finish.js]: This makes a lot of sense to me.

**Pull request merged**